lib/symboldatabase.cpp

```diff
@@ -5355,12 +5355,15 @@ void SymbolDatabase::setValueType(Token *tok, const ValueType &valuetype)
 
     if (vt1 && Token::Match(parent, "<<|>>")) {
         if (!mIsCpp || (vt2 && vt2->isIntegral())) {
-            if (vt1->type < ValueType::Type::BOOL || vt1->type >= ValueType::Type::INT)
-                setValueType(parent, *vt1);
-            else {
+            if (vt1->type < ValueType::Type::BOOL || vt1->type >= ValueType::Type::INT) {
+                ValueType vt(*vt1);
+                vt.reference = Reference::None;
```

Comment:
Thats actually a prvalue, which is correct. An xvalue is a temporary rvalue reference(such as the return to `std::move`).

Reply:
sorry you're right. I admit I find the various subtly different classes of value difficult to wrap my head around.